### PR TITLE
Support platform "any/any"

### PR DESCRIPTION
**Pull request by lucklove**
### What problem does this PR solve? 
This PR introduces the "any" platform where the packages can satisfy all platforms.

### What is changed and how it works?
In the package lookup stage, I add another step to lookup "any/any" if the previous lookup failed.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: cmd/mirror.go [hunk 1]
```diff
@@ -374,6 +379,20 @@ func newMirrorPublishCmd() *cobra.Command {
 	return cmd
 }
 
+func validatePlatform(goos, goarch string) error {
+	// Only support any/any, don't support linux/any, any/amd64 .etc.
+	if goos == "any" && goarch == "any" {
```
Comment (AstroProfundis) [inline cmd/mirror.go:384]: I don't think this bool expr is correct

File: pkg/repository/v1manifest/manifest.go [hunk 2]
```diff
@@ -47,6 +47,9 @@ const (
 	// SpecVersion of current, maybe we could expand it later
 	CurrentSpecVersion = "0.1.0"
 
+	// AnyPlatform can match any os and any arch
```
Comment (AstroProfundis) [inline pkg/repository/v1manifest/manifest.go:50]: AnyPlatform is the ID for platform independent components

File: pkg/repository/v1manifest/manifest_test.go [hunk 3]
```diff
@@ -13,5 +13,38 @@
...
 // TODO test that invalid manifests trigger errors
 // TODO test SignAndWrite
+
+func TestVersionItem(t *testing.T) {
+	manifest := &Component{
+		Platforms: map[string]map[string]VersionItem{
+			"linux/amd64": map[string]VersionItem{
+				"v1.0.0": VersionItem{Entry: "test"},
+			},
+			"any/any": map[string]VersionItem{
```
Comment (AstroProfundis) [inline pkg/repository/v1manifest/manifest_test.go:31]: should also add sth like `darwin/any`, `any/arm64`

File: cmd/mirror.go (hunk 1 above)
Comment (lucklove) [reply cmd/mirror.go:384]: It's correct. any/any is supported, any/amd64 not

**Pull request merged**